library/objective-c/EnvoyEngineImpl.m

```diff
@@ -10,6 +10,72 @@ static void ios_on_exit() {
   NSLog(@"[Envoy] library is exiting");
 }
 
+typedef struct {
+  __unsafe_unretained EnvoyHTTPFilter *filter;
+} ios_http_filter_context;
+
+// TODO: factor out conversion
+// TODO: move impl
+static envoy_data toManagedNativeString(NSString *s) {
+  size_t length = s.length;
+  uint8_t *native_string = (uint8_t *)safe_malloc(sizeof(uint8_t) * length);
+  memcpy(native_string, s.UTF8String, length);
+  envoy_data ret = {length, native_string, free, native_string};
+  return ret;
+}
+
+static EnvoyMutableHeaders *to_ios_mutable_headers(envoy_headers headers) {
```

Comment:
Out of curiosity, why does this function use snake case when the other helpers use camel case?

Reply:
Our convention has been to use snake case for functions called from the c pathway; camel case for functions called from the Objective-C pathway.

Reply:
Got it. Maybe we should document that in our contribution guide?